### DISCUSSION TITLE
Change client ip address

### DISCUSSION THREAD

**Discussion by Riivz** [PHPMailer/PHPMailer#2970] [Q&A] [answered]
Hello
I have a web application that uses PHPmailer to send notifications by email. The web application runs in docker. The mail server that I must use has strict policy and allows connection only from docker server public ip address. Every time I try send out a notification PHPmailer uses my docker local ip address as client ip address and I cant connect to server. I get this error:
[php7:notice] [pid 28] [client 172.18.0.2:44866] Connection failed. Error #2: stream_socket_client(): unable to connect to ssl://mailserver:465 (Connection refused)
172.18.0.2 is the docker local ip.
What can I do so PHPmailer uses my public ip address instead of docker local ip address for client ip address?

**Comment by Synchro** [comment, marked as answer]
PHPMailer doesn't get to choose what IP it connects from; that's set by the wider PHP configuration and the networking config in the container, in particular the default routes. This is outside PHPMailer's control.

Separately, PHPMailer doesn't like single-word hostnames because they are indistinguishable from invalid public hostnames, so you may find it's effective to rename your instance from `mailserver` to something like `mailserver.local`, or whatever is workable for your config.

**Reply by Riivz** [reply]
Thx
